output/cloud/output.go

```diff
@@ -45,6 +46,8 @@ type Output struct {
 	logger logrus.FieldLogger
 	opts   lib.Options
 
+	outv2 *expv2.Output
```

Comment:
```suggestion
	SomeCloudOutputSubInterface
```

Instead of this embedded `*expv2.Output` and all of the `if !out.config.UseVersion2.Bool` and `if out.outv2` statements sprinkled in all of the methods, it seems to me that it would be easier to add a new small `interface` type here with a few methods. Potentially a sub-set of the `output.Output` interface.

Then embed that type in the struct and have 2 implementations of that interface - one for the old json-based cloud output and one for the new protobuf one. `newOutput()` can instantiate the correct one and save it in the struct instance it returns.

Reply:
@na-- I made an attempt in the last commits, they are just the final result of the process that can be fine in the history here: https://github.com/grafana/k6/tree/cloudv2

I've created a dedicated PR for the versioning gateway so it should be clear and easier to get the architecture and merge with a different timing https://github.com/grafana/k6/pull/3041